File: app/src/main/java/lemond/annoying/gamerscompanion/module/AppContextModule.java
package lemond.annoying.gamerscompanion.module;


import android.content.Context;

import dagger.Module;
import dagger.Provides;
import lemond.annoying.gamerscompanion.GamersApplicationScope;

//provides the external module such as context
@Module
public class AppContextModule {

    private final Context context;
    // when this instances is created, context is required
    public AppContextModule(Context context) {
        this.context = context.getApplicationContext();
    }

    @Provides
    @GamersApplicationScope
    public Context provideContext() {
        return context;
    }
}

File: app/src/main/java/lemond/annoying/gamerscompanion/app/module/AppContextModule.kt
package lemond.annoying.gamerscompanion.app.module


import android.app.Application
import android.content.Context

import dagger.Module
import dagger.Provides
import lemond.annoying.gamerscompanion.app.GamersApplicationScope


@Module
class AppContextModule {

    @Provides
    @GamersApplicationScope
    internal fun provideContext(application: Application): Context {
        return application
    }

}
